File: src/main/java/pl/exsio/querydsl/entityql/examples/entity/UploadedFile.java
package pl.exsio.querydsl.entityql.examples.entity;

import javax.persistence.*;
import java.util.Arrays;
import java.util.UUID;

@Entity
@Table(name = "UPLOADED_FILES")
public class UploadedFile {

    @Id
    @Column(name = "FILE_ID", nullable = false, columnDefinition = "UUID")
    @GeneratedValue
    private UUID id;

    @Lob
    @Column(name = "DATA", nullable =  false)
    private byte[] data;

    public UploadedFile() {
    }

    public UploadedFile(byte[] data, UUID id) {
        this.id = id;
        this.data = data;
    }

    @Override
    public String toString() {
        return "UploadedFile{" +
                "id=" + id +
                ", data=" + Arrays.toString(data) +
                '}';
    }
}

File: kotlin-maven-jpa-static/src/main/kotlin/pl/exsio/querydsl/entityql/examples/jpa/entity/UploadedFile.kt
package pl.exsio.querydsl.entityql.examples.jpa.entity

import java.util.*
import javax.persistence.*

@Entity
@Table(name = "UPLOADED_FILES")
class UploadedFile() {

    @Id
    @Column(name = "FILE_ID", nullable = false, columnDefinition = "UUID")
    @GeneratedValue
    var id: UUID? = null

    @Lob
    @Column(name = "DATA", nullable = false)
    var data:ByteArray? = null

   constructor(id: UUID, data: ByteArray) : this() {
       this.id = id
       this.data = data
   }
}
